sdk/lusid/utilities/api_client_factory.py

```diff
@@ -20,23 +20,35 @@ def __init__(self, **kwargs):
         :param str api_url: LUSID API url
         :param str app_name: Application name (optional)
         :param str api_secrets_filename: Name of secrets file (including full path)
+        :param str certificate_filename: Name of the certificate file (.pem, .cer or .crt)
+        :param str proxy_url: The url of the proxy to use
         """
 
         api_client = None
         api_url = None
+        certificate_filename = None
+        proxy_url = None
 
         if "api_url" in kwargs.keys():
             api_url = kwargs["api_url"]
-
         else:
             api_url = os.getenv("FBN_LUSID_API_URL", None)
 
+        if "certificate_filename" in kwargs.keys():
+            certificate_filename = str(kwargs["certificate_filename"])
+
         if ("token" in kwargs and str(kwargs["token"]) != "None") and api_url is not None:
 
             config = lusid.Configuration()
             config.access_token = kwargs["token"]
             config.host = api_url
 
+            if certificate_filename is not None:
```

Comment:
This is duplicated in ApiClientBuilder, where it is slightly different.  Can the code be consolidated?